File: Customer-Front-End/customer-front-end/src/app/interfaces/new-order.ts
import { NewOrderItem } from "./new-order-item";

export interface NewOrder {
    restaurantNotes?: String;
    driverNotes?: String;
    subTotal: number;
    deliveryFee: number;
    tax: number;
    tip?: number;
    total: number;
    netLoyalty?: number;
    timeCreated?: Date;
    restaurantIds: number[];
    discountIds?: number[];
    items: NewOrderItem[];
}

File: Order-Microservice/order-microservice/src/main/java/com/smoothstack/ordermicroservice/data/NewOrder.java
package com.smoothstack.ordermicroservice.data;

import java.time.LocalDateTime;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class NewOrder {
    
    private String restaurantNotes;

    private String driverNotes;

    private Double subTotal;

    private Double deliveryFee;

    private Double tax;

    private Double tip;

    private Double total;

    private Integer netLoyalty;

    private LocalDateTime timeCreated;

    private List<Integer> restaurantIds;

    private List<Integer> discountIds;

    private List<NewOrderItem> items;

}
